huaweicloud/services/iam/resource_huaweicloud_identity_role_assignment.go

```diff
@@ -72,11 +115,25 @@ func resourceIdentityRoleAssignmentV3Create(ctx context.Context, d *schema.Resou
 		return fmtp.DiagErrorf("Error assigning role: %s", err)
 	}
 
-	d.SetId(buildRoleAssignmentID(domainID, projectID, groupID, roleID))
+	if val, ok := d.GetOk("project_name"); ok {
+		d.SetId(buildRoleAssignmentID(domainID, val.(string), groupID, roleID))
+	} else {
+		d.SetId(buildRoleAssignmentID(domainID, projectID, groupID, roleID))
```

Comment:
how about just keeping the before behavior, just using projectID to build the assignmentID? you already got the projectID even if project_name provided, use ID here, then you don't need to do list API request to get the project ID from name within Read/Update/Delete

Reply:
I agree